encoding/streams/decode-bad-chunks.any.js

```diff
@@ -0,0 +1,73 @@
+// META: global=worker
+
+'use strict';
+
+const badChunks = [
+  {
+    name: 'undefined',
+    value: undefined
+  },
+  {
+    name: 'null',
+    value: null
+  },
+  {
+    name: 'numeric',
+    value: 3.14
+  },
+  {
+    name: 'object, not BufferSource',
+    value: {}
+  },
+  {
+    name: 'array',
+    value: [65]
+  },
+  {
+    name: 'detached ArrayBufferView',
+    value: (() => {
+      const u8 = new Uint8Array([65]);
+      const ab = u8.buffer;
+      const mc = new MessageChannel();
+      mc.port1.postMessage(ab, [ab]);
+      return u8;
+    })()
+  },
+  {
+    name: 'detached ArrayBuffer',
+    value: (() => {
+      const u8 = new Uint8Array([65]);
+      const ab = u8.buffer;
+      const mc = new MessageChannel();
+      mc.port1.postMessage(ab, [ab]);
+      return ab;
+    })()
+  }
+];
+
+let badShared = [];
+if (typeof SharedArrayBuffer !== 'undefined') {
```

Comment:
I guess this is here so the test doesn't fail completely in user agents that have it disabled? This kind of conditional testing does feel a little bit problematic. Perhaps we can put the constructor in a callback so only the actual test fails?

Reply:
Yes. I found a simple way to make only those specific tests fail if SharedArrayBuffer is absent. PTAL.